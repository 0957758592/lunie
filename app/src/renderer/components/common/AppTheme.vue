<template lang="pug">
#app-theme
</template>

<script>
import { mapGetters } from 'vuex'
export default {
  name: 'app-theme',
  computed: { ...mapGetters(['config']) },
  methods: {
    setTheme (theme) {
      if (theme === 'light') {
        this.setThemeLight()
      } else {
        this.setThemeDark()
      }
    },
    setThemeLight () {
      this.setCssVar('app-fg', '#eee')
      this.setCssVar('app-bg', '#fff')
      this.setCssVar('app-bg-alpha', 'hsla(233, 0%, 0%, 5%)')
      this.setCssVar('bright', '#000')
      this.setCssVar('txt', '#333')
      this.setCssVar('dim', '#666')
      this.setCssVar('bc', '#ddd')
      this.setCssVar('bc-dim', '#eee')
      this.setCssVar('hover-bg', 'hsl(233, 0%, 90%)')
      this.setCssVar('input-bc', 'hsl(233, 22%, 67%)')
      this.setCssVar('input-bc-hover', 'hsl(233, 22%, 40%)')
    },
    setThemeDark () {
      this.setCssVar('app-fg', 'hsl(233, 33%, 16%)')
      this.setCssVar('app-bg', 'hsl(233, 36%, 13%)')
      this.setCssVar('app-bg-alpha', 'hsla(233, 36%, 13%, 95%)')
      this.setCssVar('bright', '#fff')
      this.setCssVar('txt', 'hsl(233, 13%, 85%)')
      this.setCssVar('dim', 'hsl(233, 13%, 60%)')
      this.setCssVar('bc', 'hsl(233, 22%, 23%)')
      this.setCssVar('bc-dim', 'hsl(233, 33%, 16%)')
      this.setCssVar('hover-bg', 'hsl(233, 43%, 10%)')
      this.setCssVar('input-bc', 'hsl(233, 22%, 33%)')
      this.setCssVar('input-bc-hover', 'hsl(233, 22%, 40%)')
    },
    setCssVar (key, value) {
      document.documentElement.style.setProperty(`--${key}`, value)
    }
  },
  mounted () {
    this.setTheme(this.config.theme)
  },
  watch: {
    'config.theme' (newTheme) {
      this.setTheme(newTheme)
    }
  }
}
</script>
