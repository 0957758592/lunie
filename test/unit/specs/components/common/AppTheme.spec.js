import AppTheme from 'common/AppTheme'
import htmlBeautify from 'html-beautify'
import setup from '../../../helpers/vuex-setup'

describe('AppTheme', () => {
  let wrapper, store, instance
  let { mount } = setup()

  beforeEach(() => {
    instance = mount(AppTheme)
    wrapper = instance.wrapper
  })

  it('has the expected html structure', () => {
    expect(htmlBeautify(wrapper.html())).toMatchSnapshot()
  })

  /*
  it('sets the theme on mount', () => {
    const spy = jest.spyOn(wrapper.vm, 'setTheme')
    mount(AppTheme)
    expect(spy).toHaveBeenCalled()
  })
  */

  it('can change change the theme', () => {
    expect(wrapper.vm.setTheme).toBeTruthy()
  })

  it('can change the theme to dark', () => {
    expect(wrapper.vm.setThemeDark).toBeTruthy()
  })

  it('can change the theme to light', () => {
    expect(wrapper.vm.setThemeLight).toBeTruthy()
  })

  it('can set a css variable', () => {
    expect(wrapper.vm.setCssVar).toBeTruthy()
  })
})
