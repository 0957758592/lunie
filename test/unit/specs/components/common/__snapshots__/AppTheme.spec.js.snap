// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTheme has the expected html structure 1`] = `"<div id=\\"app-theme\\"></div>"`;
